### PR TITLE
Change cleanPositions to static method

### PR DESCRIPTION
**Pull request by studiokiwik**
Change all method cleanPosition to a staic method
change all call to this method to a static call

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
Hi @studiokiwik, thank for contribution.

We will review it asap

File: classes/Attribute.php [hunk 1]
```diff
@@ -339,7 +339,7 @@ public function updatePosition($way, $position)
      * @param bool $use_last_attribute
      * @return bool $return
      */
-    public function cleanPositions($id_attribute_group, $use_last_attribute = true)
+    public static function cleanPositions($id_attribute_group, $use_last_attribute = true)
```
Comment (PrestaEdit) [inline classes/Attribute.php]: You can't do it here. There is : (int)$this->id inside the method.

**Comment by mickaelandrieu** [comment]
ping @Shudrum, it's an API change: what do you think ?

**Comment by xBorderie** [comment]
Thank you for this PR!

Could you amend your commits' names to use the [guidelines](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message), so that it is taken into account by our changelog tool?

To do this, open a command line window and use `git commit --amend` for the commit's name. See [GitHub's help page](https://help.github.com/articles/changing-a-commit-message/) for more information.

_Note: this **must** be done via the command line: you can't do this just by changing the title of the pull-request from the GitHub interface!_ :)

Thank you!

**Comment by maximebiloe** [comment]
Hello @studiokiwik,

Thanks for your contribution.
Can you tell us the reason why you need the method to be static?

Also, could you squash your commits so there will be only one, please?

Regards

**Comment by studiokiwik** [comment]
Hello @maximebiloe 

Half of the methods cleanPositions was static, we've decided to use static everywhere to maintain cohesion with methods which are called without parameters (for example in Carrier, Feature, AttributeGroup ...), as they are modifying positions of all objects.
We could refactor all as dynamic methods, but I think it would be less universal, and unnecessary as there is probably no need to initialize whole object to make position cleanup (unless someone wish to add a hook)

Also, generally speaking, arguments to that method should be optional by default, which will allow to perfom maintenace operations more easily.

**Comment by maximebiloe** [comment]
Thank you for your answer.
@Shudrum, what do you think about this?

File: classes/Attribute.php [hunk 2]
```diff
@@ -339,13 +339,13 @@ public function updatePosition($way, $position)
      * @param bool $use_last_attribute
      * @return bool $return
      */
-    public function cleanPositions($id_attribute_group, $use_last_attribute = true)
+    public static function cleanPositions($id, $id_attribute_group, $use_last_attribute = true)
```
Comment (aleeks) [inline classes/Attribute.php]: We can't do this now, 1.7 has been released so now, if we do this, retro-compatibility will be broken :/

**Comment by aleeks** [comment]
Hello @studiokiwik,

In relation to my comment, i'm closing this PR because we can't do this right now. (and i know it's our fault)

> Half of the methods cleanPositions was static, we've decided to use static everywhere to maintain cohesion with methods which are called without parameters (for example in Carrier, Feature, AttributeGroup ...), as they are modifying positions of all objects. >

I'm totaly agree with your argument because it also annoys me when I see it:

kiwi

It could be interesting to keep that in mind for a next major release, so i'm closing this PR ... :/

**Pull request closed without merging**